crates/spirv-std/macros/src/lib.rs

```diff
@@ -67,38 +67,40 @@ use quote::ToTokens;
 ///
 /// The grammar for the macro is as follows:
 ///
-/// ```no_compile
+/// ```rust,no_compile
 /// Image!(
 ///     <dimensionality>,
-///     <type|format>,
+///     <type=...|format=...>,
 ///     [sampled[=<true|false>],]
 ///     [multisampled[=<true|false>],]
 ///     [arrayed[=<true|false>],]
 ///     [depth[=<true|false>],]
 /// )
 /// ```
 ///
+/// `=true` can be omitted as shorthand - e.g. `sampled` is short for `sampled=true`.
+///
 /// A basic example looks like this:
-/// ```no_compile
+/// ```rust,no_compile
 /// #[spirv(vertex)]
 /// fn main(#[spirv(descriptor_set = 0, binding = 0)] image: &Image!(2D, type=f32, sampled)) {}
 /// ```
 ///
 /// ## Arguments
 ///
-/// - `dimensionality` — Dimensionality of an image. Accepted values: `1D`,
-///   `2D`, `3D`, `rect`, `cube`, `subpass`.
+/// - `dimensionality` — Dimensionality of an image.
+///    Accepted values: `1D`, `2D`, `3D`, `rect`, `cube`, `subpass`.
 /// - `type` — The sampled type of an image, mutually exclusive with `format`,
-///   when set the image format is unknown. Accepted values: `f32`, `f64`,
-///   `u8`, `u16`, `u32`, `u64`, `i8`, `i16`, `i32`, `i64`.
+///    when set the image format is unknown.
+///    Accepted values: `f32`, `f64`, `u8`, `u16`, `u32`, `u64`, `i8`, `i16`, `i32`, `i64`.
 /// - `format` — The image format of the image, mutually exclusive with `type`.
-///   Accepted values: Camel case versions of [`ImageFormat`].
-/// - `sampled` — Whether it is known that the image will be used with a sampler
-///   at compile time. Accepted values: `true` or `false`. Default: `unknown`.
-/// - `multisampled` — Whether the image contains multisampled content. Accepted
-///    values: `true` or `false`. Default: `false`.
-/// - `arrayed` — Whether the image contains arrayed content. Accepted
-///    values: `true` or `false`. Default: `false`.
+///    Accepted values: Camel case versions of [`ImageFormat`].
```

Comment:
[These](https://github.com/EmbarkStudios/rust-gpu/blob/c3eff4d4a5a52607aa3bc8b4ba8409f6659425ee/crates/spirv-std/macros/src/image.rs#L372-L412) don't look camel case to me